quantum/nodes/toffoli-gate/toffoli-gate.js

```diff
@@ -0,0 +1,123 @@
+        );
+      } else if (typeof msg.payload.qubit === "undefined") {
+        throw new Error(
+          'If "Registers & Bits" was selected in the "Quantum Circuit" node, please make use of register nodes.'
+        );
+      }
+
+      // Store all the qubit objects received as input into the node.qubits array
+      node.qubits.push(msg);
+
+      // If all qubits have arrived, we first reorder the node.qubits array for output consistency
+      if (node.qubits.length == 3) {
+        node.qubits.sort(function compare(a, b) {
+          if (typeof a.payload.register !== "undefined") {
+            const regA = parseInt(a.payload.registerVar.slice(2));
+            const regB = parseInt(b.payload.registerVar.slice(2));
+            if (regA < regB) return -1;
+            else if (regA > regB) return 1;
+            else {
+              if (a.payload.qubit < b.payload.qubit) return -1;
+              else return 1;
+            }
+          } else {
+            if (a.payload.qubit < b.payload.qubit) return -1;
+            else return 1;
+          }
+        });
+
+        //Initialise qubit variables for script.
+        let control1 = node.qubits[0];
+        let control2 = node.qubits[0];
+        let target = node.qubits[0];
+
+        //Determine which node is the target based on position.
+        if (node.targetPosition == "Top") {
+          target = node.qubits[0];
+          control1 = node.qubits[1];
+          control2 = node.qubits[2];
+        } else if (node.targetPosition == "Middle") {
+          target = node.qubits[1];
+          control1 = node.qubits[0];
+          control2 = node.qubits[2];
+        } else {
+          target = node.qubits[2];
+          control1 = node.qubits[0];
+          control2 = node.qubits[1];
+        }
+
+        // Generate the corresponding Toffoli Gate Qiskit script
+        let toffoliCode = "";
+        node.qubits.map((msg) => {
+          //Use qubits only if there are no registers.
+          if (typeof msg.payload.register === "undefined") {
+            toffoliCode = util.format(
+              snippets.TOFFOLI_GATE,
+              control1.payload.qubit.toString(),
+              control2.payload.qubit.toString(),
+              target.payload.qubit.toString()
+            );
+          } else {
+            //Use registers if there are quantum registers.
+            toffoliCode = util.format(
+              snippets.TOFFOLI_GATE,
+              control1.payload.registerVar +
+                "[" +
+                control1.payload.qubit.toString() +
+                "]",
+              control2.payload.registerVar +
+                "[" +
+                control2.payload.qubit.toString() +
+                "]",
+              target.payload.registerVar +
+                "[" +
+                target.payload.qubit.toString() +
+                "]"
+            );
+          }
+        });
+
+        console.log(toffoliCode);
```

Comment:
Get rid of console log and that's all good for me

Reply:
Thanks for pointing that out, Theo! I've removed it.